lib/plenario2/core/changesets/virtual_point_field_changesets.ex

```diff
@@ -1,53 +1,35 @@
-defmodule Plenario2.Core.Changesets.VirtualPointFieldChangeset do
+defmodule Plenario2.Core.Changesets.VirtualPointFieldChangesets do
   import Ecto.Changeset
 
-  def create(struct, params) do
+  def create_from_loc(struct, params) do
     struct
-    |> cast(params, [:longitude_field, :latitude_field, :location_field, :meta_id])
-    |> validate_required([:meta_id])
-    |> _validate_longlat_or_loc()
+    |> cast(params, [:location_field, :meta_id])
+    |> validate_required([:location_field, :meta_id])
     |> cast_assoc(:meta)
-    |> _set_name()
+    |> _set_name_loc()
+  end
+
+  def create_long_lat(struct, params) do
```

Comment:
Maybe have this named `create_from_long_lat` to be consistent with `create_from_loc`? When I look at `create_long_lat` my first thought is that you're creating longitude and latitude coordinates, not that you're creating a struct from them.